### PR TITLE
refactor: remove dead branch in artifact_cleanup

### LINKED ISSUE #279: [Maintenance] Remove dead code in artifact_cleanup.py
## Summary

The `has_partial_overlap_prev` function in `pytrendy/post_processing/segments_refine/artifact_cleanup.py` contains a branch (lines 113-114) that was never triggered by any test data. This is dead code that should be removed.

## Background

This code was added by **Copilot** in commit `9148d8c` (PR #46):
```
fix(test): add automated tests for noise crashes and edge cases for full code coverage, refactor clean artifacts, and resolve mpl test bugs (#46)
```

The function checks whether a trend segment overlaps with a previous Flat segment of similar size. The comment references "edge case scenario 11" (suggesting `noisy_edgecase_11` data would trigger it), but that data never produces overlapping segments.

This has likely to have had covered it at one point - but now since the latest bug fix changed key behaviour on distriguishing between trends and flats (overall net improvement) it seems like its broken this previously covered line of code.

## The dead branch

```python
def has_partial_overlap_prev(segment, segment_prev):
    ...
    if is_overlap_prev and (is_trend and (is_prev_noise or is_prev_opposite_trend) and is_curr_shorter):
        return True  # ← This branch IS reachable
    if is_overlap_prev and (is_trend and is_prev_flat) and is_curr_similar:
        return True  # ← This branch (lines 113-114) is NEVER reached
    return False
```

## Why it's unreachable

`detect_trends` produces non-overlapping segments. The overlap condition (`start <= prev_end`) requires the current segment to start before the previous segment ends, which the pipeline never produces. The function always returns `False` at line 115 in practice.

## Coverage evidence

- `has_partial_overlap_prev` line 115 (`return False`) IS covered
- Lines 113-114 (`return True` for the Flat branch) are NEVER covered
- This was true on `develop`, on `nodates`, and across all test suites

## Action

Remove the unreachable branch (lines 113-114) or the entire `has_partial_overlap_prev` function if no callers depend on it. Or consider changing the linked test / creating a new one to hit it (if possible)

## Related

- PR #46 (commit `9148d8c`) — original Copilot-authored code
- PR #224 — flat detection fix that changed key behaviour on distinguishing between trends and flats
- PR #278 — coverage work that confirmed this is dead code
- Line 115: `artifact_cleanup.py` — 1 remaining uncovered line in 99.9% coverage

### PR DESCRIPTION
**Pull request by RussellSB**
## Summary

Removes the unreachable branch in `artifact_cleanup.has_overlap_prev` (the "edge case scenario 11" trend-after-flat path), tracked in #279. The branch was written aspirationally by Copilot PR #46, but the pipeline never produces overlapping trend/flat neighbours, so it could never fire. Closes #279.

## Changes

- Deleted the `is_overlap_prev and (is_trend and is_prev_flat) and is_curr_similar` branch (the covered noise/opposite-trend branch above it is untouched).
- Removed the now-unused conditions `is_curr_similar` / `is_prev_flat` and the stale `TODO: Cleanup redundant condition statements` comment.

## Verification

- Full suite: **165 passed / 0 failed**.
- Coverage: **100%** (1154/1154 statements — zero uncovered lines; this branch was the last miss).

## Unreachability argument

Coverage alone proves the branch untested, not unreachable — the stronger argument: a sweep of the entire test corpus (every fixture x column x params combination in `tests/` and `pytrendy/io/data/`) plus all recorded detection runs never produced a Flat segment immediately followed by an overlapping same-size trend (`is_overlap_prev and is_trend and is_prev_flat and is_curr_similar`), because detection alternates direction at turning points. The branch was written aspirationally by Copilot PR #46 for a "scenario 11" that no dataset reaches, as documented in the issue.

## For review

- Actioned: `.codecov.yml` now enforces a 100% floor (project + patch) in the latest commit, so coverage cannot degrade going forwards.

### CODE REVIEW

**Comment by RussellSB** [comment]
Yes feel free to hard enforce code coverage at 100% going forwards now

File: pytrendy/post_processing/segments_refine/artifact_cleanup.py [hunk 1]
```diff
@@ -97,20 +97,16 @@ def has_overlap_prev(segment: dict, segment_prev: dict) -> bool:
...
-        is_curr_similar = (prev_width <= 1.5 * width) and (prev_width >= 0.5 * width)
 
         is_trend = (dir in ('Up', 'Down'))
         is_prev_noise = (prev_dir == 'Noise')
         is_prev_opposite_trend = (prev_dir in ('Up', 'Down') and prev_dir != dir)
-        is_prev_flat = (prev_dir == 'Flat')
 
         if is_overlap_prev and (is_trend and (is_prev_noise or is_prev_opposite_trend) and is_curr_shorter):
             return True # overlap when curr is trend and prev is noise of larger/equal window
-        if is_overlap_prev and (is_trend and is_prev_flat) and is_curr_similar:
```
Comment (RussellSB) [inline pytrendy/post_processing/segments_refine/artifact_cleanup.py:112]: so this achieves 100% code coverage, can you expand on why this is never hit? And remind me why this was initially added and at what point it became outdated.

With respect to which test scenarios was it related?

Comment (RussellSB) [reply pytrendy/post_processing/segments_refine/artifact_cleanup.py:112]: [opencode-go/glm-5.3-flash]: Full archaeology on all three questions:

**Why it was added** — the condition predates the PR that named it: introduced in PR #28 (`1e59cab`, Oct 2025) as *"checking for edge case of abrupt refined overlapping flat. Delete trend when a weak borderline case of similar length to overlapping flat"* (original wording: *"saw this in noise 20 edge case"*). PR #33 (`3d4aa64f`) carried it into `artifact_cleanup.py`; Copilot PR #46 (`ad70081`) relocated it into `has_overlap_prev`, added the `noisy_edgecase_11` column + `test_noisy_edgecase_11_scenario`, and relabelled it "edge case scenario 11".

**Which test scenarios** — only `noisy_edgecase_11` ever exercised it (`test_noisy_edgecase_11_scenario`).

**Why it became never-hit** — it actually fired continuously until PR #177 (`e01fdce5`, Jun 2026), whose zero-baseline fix resets `segment_length = 0` on min-length failure in `get_segments`. Its last firing pair was a Down `2025-03-30→04-06` (width 7) overlapping a Flat `2025-03-22→03-31` (width 9) — exactly `is_overlap_prev ∧ is_trend ∧ is_prev_flat ∧ is_curr_similar`. After that fix the region forms as Flat `03-22→03-28` + Noise `03-29→04-09`, so the overlapping Down never re-forms (issue #279 blamed PR #224, but it was already dead before #224 merged).

**Structural reason** — detection emits contiguous, non-overlapping segments; cleanup's transient overlaps are expand/contract and abrupt-shaving artefacts between trend↔noise/opposite-trend, and Flats are never extended by those steps. The trend-over-Flat overlap needed that specific pre-#177 bleed, and `is_curr_similar` is a narrow ±50% window — today the Flat before the `noisy_edgecase_11` Up is 2 days vs 25, so both predicates fail.

Timeline: PR #28 (introduced) → PR #33 (moved) → PR #46 (relocated + named) → PR #177 (became unreachable) → #279 (the suite's sole uncovered line) → PR #310 (removed).
